### DISCUSSION TITLE
Combine/use multiple topics?

### DISCUSSION THREAD

**Discussion by darkspirit510** [arachnetech/homebridge-mqttthing#665] [Q&A]
Hi all! Maybe someone knows if the following is possible without external software.

I want to add my garage door opener to homebridge. According to documentation, the door opener topic `getCurrentDoorState` reports one of five positions (start/end/opening/closing/stopped). My garage door controller has the following topics (opened, closed, opening, closing). In other words - I need to merge four different topics to calculate current door state. Is this possible with mqttthing?

Thank you!

Sascha (darkspirit510)

**Comment by darkspirit510** [comment]
Does not seem to be possible, there's an open issue https://github.com/arachnetech/homebridge-mqttthing/issues/170. Closing the question then.

**Comment by NorthernMan54** [comment]
Take a look at node-red and the package HomeKit-bridged, it allows the creation of custom HomeKit devices. The initial learning curve is huge, but it would do what your looking for.
